.changeset/changelog-stylelint.js

```diff
@@ -5,9 +5,44 @@
 const { getInfo, getInfoFromPullRequest } = require('@changesets/get-github-info');
 
 /**
- * @type {import('@changesets/types').ChangelogFunctions}
+ * @type {Array<string | undefined>}
+ */
+const CHANGESET_SECTIONS = ['Deprecated', 'Removed', 'Changed', 'Added', 'Fixed'];
+
+const changesetSectionReg = new RegExp(`^(${CHANGESET_SECTIONS.join('|')}): `);
+
+/**
+ * @typedef { 'major' | 'minor' | 'patch' } ReleaseType
+ * @typedef { Record<ReleaseType, Array<Promise<string>>> } ReleaseLines
+ * @typedef { Record<ReleaseType, string[]> | string[] } ResolvedReleaseLines
+ * @type {import('@changesets/types').ChangelogFunctions & { reorderReleaseLines(releaseLines: ReleaseLines): Promise<ResolvedReleaseLines> }}
  */
 const changelogFunctions = {
+	async reorderReleaseLines(releaseLines) {
+		const resolved = {
+			major: await Promise.all(releaseLines.major),
+			minor: await Promise.all(releaseLines.minor),
+			patch: await Promise.all(releaseLines.patch),
+		};
+
+		return Object.entries(resolved)
+			.reduce((acc, [_type, lines]) => {
+				const type = /**  @type {ReleaseType} */ (_type);
+				lines.forEach((line) => {
+					if (line) {
+						acc.push(`${line}${type === 'major' ? ' (BREAKING)' : ''}`);
+					}
+				});
+				return acc;
+			}, /**  @type {string[]} */ ([]))
+			.sort((a, b) => {
+				const aSection = changesetSectionReg.exec(a)?.[0];
+				const bSection = changesetSectionReg.exec(b)?.[0];
+				return aSection === bSection
+					? b.localeCompare(a)
+					: CHANGESET_SECTIONS.indexOf(bSection) - CHANGESET_SECTIONS.indexOf(aSection);
```

Comment:
[sugget] The order seems incorrect:

```suggestion
					? a.localeCompare(b)
					: CHANGESET_SECTIONS.indexOf(aSection) - CHANGESET_SECTIONS.indexOf(bSection);
```